### PR TITLE
Update @rollup/plugin-node-resolve

### PR DESCRIPTION
**Pull request by hsubox76**
Update @rollup/plugin-node-resolve dependency as attempted by this Renovate PR: https://github.com/firebase/firebase-js-sdk/pull/4162

Requires one small fix due to breaking change described here: https://github.com/rollup/plugins/pull/656

### CODE REVIEW

File: common/api-review/performance-exp.api.md [hunk 1]
```diff
@@ -13,10 +13,7 @@ export interface FirebasePerformance {
 }
 
 // @public
-export function getPerformance(app: FirebaseApp): FirebasePerformance;
-
-// @public
-export function initializePerformance(app: FirebaseApp, settings?: PerformanceSettings): FirebasePerformance;
+export function getPerformance(app: FirebaseApp, settings?: PerformanceSettings): FirebasePerformance;
```
Comment (Feiyang1) [inline common/api-review/performance-exp.api.md:16]: seems unrelated to this PR. and it's overwriting a newer version of the api report

File: repo-scripts/size-analysis/bundle/rollup.ts [hunk 2]
```diff
@@ -36,9 +36,7 @@ export async function bundleWithRollup(
   };
 
   if (moduleDirectory) {
-    resolveOptions.customResolveOptions = {
-      moduleDirectory
-    };
+    resolveOptions.moduleDirectories = [moduleDirectory];
```
Comment (Feiyang1) [inline repo-scripts/size-analysis/bundle/rollup.ts:39]: Nice! How did you catch the problem?

**Review by Feiyang1 — APPROVED** [review]
LGTM. Please remove the api reports.

File: repo-scripts/size-analysis/bundle/rollup.ts (hunk 2 above)
Comment (hsubox76) [reply repo-scripts/size-analysis/bundle/rollup.ts:39]: I mean, it errors if you try to run it with the upgraded plugin, so I checked the changelog for breaking changes and this was there.

File: common/api-review/analytics-exp.api.md [hunk 3]
```diff
@@ -37,7 +37,7 @@ export type CustomEventName<T> = T extends EventNameString ? never : T;
 // @public
 export interface CustomParams {
     // (undocumented)
-    [key: string]: unknown;
+    [key: string]: any;
```
[the remark this answers is not in the document]
Comment (hsubox76) [reply common/api-review/analytics-exp.api.md:40]: Reverted.

File: common/api-review/performance-exp.api.md (hunk 1 above)
Comment (hsubox76) [reply common/api-review/performance-exp.api.md:16]: Reverted.
